src/lib/profiles/data-management/Current/LoggingManagement.cpp

```diff
@@ -436,6 +437,210 @@ void LoggingManagement::DestroyLoggingManagement(void)
     Platform::CriticalSectionExit();
 }
 
+/**
+ * Persist the Weave events of all importance types.
+ *
+ * Serializes the events in WeaveCircularTLVBuffer and the associated states into the
+ * supplied buffer and set mShutdownInProgress flag to true such that no further
+ * events can be added to the buffer.
+ *
+ * This method is intended to be used by devices that do not retain RAM while sleeping,
+ * allowing them to persist events before going to sleep and thereby prevent lost of
+ * events
+ */
+WEAVE_ERROR LoggingManagement::PersistEvent(uint8_t * buf, uint32_t bufSize, uint32_t & serializedLen)
```

Comment:
An event-by-event blit seem highly in-inefficient and un-necessary - why aren't we just global copy/write of the entire buffer contents (ptr + size)?